commons-web/src/main/java/eu/europeana/api/commons/service/authorization/AuthorizationService.java

```diff
@@ -1,16 +1,42 @@
-package eu.europeana.api.commons.service.authorization;
-
-import javax.servlet.http.HttpServletRequest;
-
-import eu.europeana.api.commons.web.exception.ApplicationAuthenticationException;
-
-public interface AuthorizationService {
-
-    /**
-     * Processes the HTTP request and validates the provided APIKey (see also Europeana APIKEY service) 
-     * @param request api request
-     * @throws ApplicationAuthenticationException if the APIKey was not submitted with the request or the APIKey could not be validated 
-     */
-    public void authorizeReadAccess(HttpServletRequest request)
-		throws ApplicationAuthenticationException;
-}
+package eu.europeana.api.commons.service.authorization;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.security.core.Authentication;
+
+import eu.europeana.api.commons.exception.ApiKeyExtractionException;
+import eu.europeana.api.commons.exception.AuthorizationExtractionException;
+import eu.europeana.api.commons.web.exception.ApplicationAuthenticationException;
+
+public interface AuthorizationService {
+
+    /**
+     * Processes the HTTP request and validates the provided APIKey (see also Europeana APIKEY service) 
+     * @param request api request
+     * @throws ApplicationAuthenticationException if the APIKey was not submitted with the request or the APIKey could not be validated 
+     */
+    public void authorizeReadAccess(HttpServletRequest request)
+		throws ApplicationAuthenticationException;
+
+    /**
+     * This method adopts KeyCloack token from HTTP request
+     * @param request The HTTP request
+     * @return list of Authentication objects
+     * @throws ApplicationAuthenticationException
+     * @throws ApiKeyExtractionException
+     */
+    public List<? extends Authentication> processJwtToken(HttpServletRequest request) 
```

Comment:
we don#t need this method in the interface

Reply:
processJwtToken method removed from interface